tests/smoke/events.js

```diff
@@ -81,11 +83,24 @@
 					shiftKey: false
 				}));
 
-				// wait for spellcheck to fire after the spacebar
+				resumeAfter(editor, 'spellCheckComplete', function () {
```

Comment:
I think this would be much easier to read if the anonymous check function defined here was instead defined outside the `resumeAfter` and given a name (e.g. `assertNoAjaxEventAndRootIsSecondParagraph`), then passed to `resumeAfter`.

For the parent anonymous function, maybe it's called `assertAjaxEventThenTriggerSpacebar`.

Is there any concept of chained promises available to avoid these nested callbacks?